src/compiler/scala/tools/nsc/javac/JavaParsers.scala

```diff
@@ -802,12 +820,67 @@ trait JavaParsers extends ast.parser.ParsersCommon with JavaScanners {
           javaLangObject()
         }
       val interfaces = interfacesOpt()
-      val (statics, body) = typeBody(CLASS, name)
+      val (statics, body) = typeBody(CLASS)
       addCompanionObject(statics, atPos(pos) {
         ClassDef(mods, name, tparams, makeTemplate(superclass :: interfaces, body))
       })
     }
 
+    def recordDecl(mods: Modifiers): List[Tree] = {
```

Comment:
Just checking: do we need to include the generated private fields? I assume that since they are private we don't need to.

Reply:
We don't need to include them 👍